### PR TITLE
server: skip null config zones in diagnostic report

### LINKED ISSUE #24499: server: null `config` in `system.zones` row crashes diagnostic report generation
https://sentry.io/cockroach-labs/cockroachdb/issues/515287016/

```
*log.safeError: stopper.go:175: *runtime.TypeAssertionError: interface conversion: tree.Datum is tree.dNull, not *tree.DBytes
  File "github.com/cockroachdb/cockroach/pkg/server/updates.go", line 323, in getReportingInfo
  File "github.com/cockroachdb/cockroach/pkg/server/updates.go", line 392, in reportDiagnostics
  File "github.com/cockroachdb/cockroach/pkg/server/updates.go", line 249, in maybeReportDiagnostics
  File "github.com/cockroachdb/cockroach/pkg/server/updates.go", line 115, in func1
  File "github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go", line 192, in func1
...
(4 additional frame(s) were not displayed)

stopper.go:175: *runtime.TypeAssertionError: interface conversion: tree.Datum is tree.dNull, not *tree.DBytes
```

### PR DESCRIPTION
**Pull request by dt**
Fixes #24499.

This is intended for back port to 2.0 as well /cc @cockroachdb/release

### CODE REVIEW

File: pkg/server/updates_test.go [hunk 1]
```diff
@@ -161,6 +161,9 @@ func TestReportUsage(t *testing.T) {
 			t.Fatalf("error applying zone config %q to %q: %v", cmd.config, cmd.resource, err)
 		}
 	}
+	if _, err := db.Exec(`INSERT INTO system.zones (id, config) VALUES (10000, null)`); err != nil {
```
Comment (a-robinson) [inline pkg/server/updates_test.go]: We should also add a test for a null ID while we're here.

**Review by a-robinson — APPROVED** [review]
LGTM, sorry for not handling this in the first place. I still don't know how someone got a null config unless they manually wrote to the `system.zones` table, but that's no excuse for not dealing with it.

File: pkg/server/updates_test.go (hunk 1 above)
Comment (dt) [reply pkg/server/updates_test.go]: I don't think that's possible: `id     INT PRIMARY KEY,`

Comment (a-robinson) [reply pkg/server/updates_test.go]: Duh, of course. Nevermind.
